astropy/io/misc/hdf5.py

```diff
@@ -276,6 +311,21 @@ def write_table_hdf5(table, output, path=None, compression=False,
         else:
             raise OSError("Table {0} already exists".format(path))
 
+    # Encode any mixin columns as plain columns + appropriate metadata
+    table = _encode_mixins(table)
+
+    # Warn if information will be lost when serialize_meta=False.  This is
+    # hardcoded to the set difference between column info attributes and what
+    # HDF5 can store natively (name, dtype) with no meta.
+    if serialize_meta is False:
+        for col in table.itercols():
+            for attr in ('unit', 'format', 'description', 'meta'):
+                if getattr(col.info, attr, None) not in (None, {}):
+                    warnings.warn("table contains column(s) with defined 'unit', 'format',"
+                                  " 'description', or 'meta' info attributes. These will"
+                                  " be dropped since serialize_meta=False.",
+                                  AstropyUserWarning)
```

Comment:
maybe add a `break` here to jump out of the loop?
Though the test suggests one gets only one warning anyway?

Reply:
Fixed.